### PR TITLE
feat(telemetry): Pushed changes to optimise the delay after the process end…

### PR DESCRIPTION
**Pull request by rupareddy5-21**
…s and to calculate configfileOptions

### CODE REVIEW

File: src/cli/commands/start/start.ts [hunk 1]
```diff
@@ -92,7 +92,7 @@ export async function start(options: SWACLIConfig) {
     else if (!fs.existsSync(outputLocation!)) {
       logger.error(`The folder "${resolvedOutputLocation}" is not found. Exit.`, true);
       const cmdEndTime = new Date().getTime();
-      collectTelemetryEvent(TELEMETRY_EVENTS.Start, {
+      await collectTelemetryEvent(TELEMETRY_EVENTS.Start, {
```
Comment (sulabh-msft) [inline src/cli/commands/start/start.ts:95]: so collectTelemetryEvent is async call

File: src/core/utils/options.ts [hunk 2]
```diff
@@ -6,6 +6,32 @@ import { logger } from "./logger";
...
 let userDefinedOptions: SWACLIConfig = {};
 let configFileDefinedOptions: SWACLIConfig = {};
+const commandSpecificOptions = {
+  init: ["yes"],
+  start: [
+    "appLocation",
+    "outputLocation",
+    "apiLocation",
+    "appDevserverUrl",
+    "apiDevserverUrl",
```
Comment (sulabh-msft) [inline src/core/utils/options.ts:16]: not sure if this scalable.. is there a way to build this list dynamically from the these commands code?

File: src/core/telemetry/baseTelemetrySender.ts [hunk 3]
```diff
@@ -90,6 +90,7 @@ export class BaseTelemetrySender implements TelemetrySender {
         this._telemetryClient = client;
         this._instantiationStatus = InstantiationStatus.INSTANTIATED;
         this._flushQueues();
+        this._telemetryClient.flush();
```
Comment (sulabh-msft) [inline src/core/telemetry/baseTelemetrySender.ts:93]: flush is async .. do we need to await() ?

**Comment by Timothyw0** [comment]
Hi @rupareddy5-21, thank you for your PR and apologies on the delay of response. I see that this PR is opened to merge into your own forked repo so I will be closing this for now. Please do not hesitate to raise a new PR to the static-web-apps-cli repo for your fix. 

Thank you again!

**Pull request closed without merging**